src/core/server/plugins/types.ts

```diff
@@ -24,7 +24,12 @@ import { ConfigPath, EnvironmentMode, PackageInfo } from '../config';
 import { LoggerFactory } from '../logging';
 import { CoreSetup, CoreStart } from '..';
 
-export type PluginConfigSchema = Type<unknown> | null;
+export type PluginConfigSchema<T = unknown> = Type<T>;
+
+export interface PluginConfigDescriptor<T = any> {
+  exposeToBrowser?: Array<keyof T>;
+  schema: PluginConfigSchema<T>;
+}
```

Comment:
The `<T =any>` here instead of `unknow` allows the plugins to use it either with or without specifying the type
```typescript
export const config: PluginConfigDescriptor = {
  exposeToBrowser: ['uiProp'],
  schema: configSchema,
};

export const config: PluginConfigDescriptor<ConfigType> = {
  exposeToBrowser: ['uiProp'],
  schema: configSchema,
};
```
with default unknown type, the `exposeToBrowser?: Array<keyof T>;` will be an `Array<never>`, forcing to actually specify the type when declaring the `PluginConfigDescriptor`